srcpkgs/ngspice/template

```diff
@@ -1,28 +1,62 @@
 # Template file for 'ngspice'
 pkgname=ngspice
-version=33
+version=35
 revision=1
 build_style=gnu-configure
-configure_args="--with-readline=yes --with-ngshared --enable-xspice --enable-cider"
+configure_args="--with-readline=yes --enable-xspice --enable-cider"
+make_build_args="-C .build_lib"
 hostmakedepends="bison byacc"
 makedepends="readline-devel libX11-devel libXaw-devel"
+checkdepends="perl"
 short_desc="Mixed Mode Mixed Level Circuit Simulator based on Spice3F5"
 maintainer="Toyam Cox <Vaelatern@voidlinux.org>"
 license="BSD-3-Clause"
 homepage="http://ngspice.sourceforge.net"
 distfiles="${SOURCEFORGE_SITE}/ngspice/ng-spice-rework/${version}/${pkgname}-${version}.tar.gz"
-checksum=b99db66cc1c57c44e9af1ef6ccb1dcbc8ae1df3e35acf570af578f606f8541f1
+checksum=c1b7f5c276db579acb3f0a7afb64afdeb4362289a6cab502d4ca302d6e5279ec
+
+# Configure can handle building ngspice as a library or as a binary, but not both at once.
+# --with-ngshared builds the library, but not binary, and readline should not be configured for the library,
+# see: https://github.com/imr/ngspice/commit/b86c85f85bbba6e45dc030af3e853edf8b9cfa3d
+pre_configure() {
+	install -d .build_lib
+	cd .build_lib
+	../configure --with-ngshared ${configure_args/--with-readline=yes/}
+}
+
+# library built during first do_build, now build the binary
+post_build() {
+	make_build_args= do_build
+}
```

Comment:
I think it's better to remove `--with-readline=yes` from `configure_args` then
```suggestion
do_configure() {
	mkdir build-lib
	cd build-lib
	../configure --with-ngshared ${configure_args}
	mkdir build
	../configure --with-readline=yes ${configure_args}
}

do_build() {
	make -C build-lib ${makejobs}
	make -C build ${makejobs}
}
```

Reply:
Applied suggestion, with just minor tweaks